homeassistant/components/unifi/switch.py

```diff
@@ -112,6 +108,18 @@ def items_added(
     items_added()
     known_poe_clients.clear()
 
+    @callback
+    def async_add_outlet_switch(_: ItemEvent, obj_id: str) -> None:
+        """Add power outlet switch from UniFi controller."""
+        if not controller.api.outlets[obj_id].has_relay:
+            return
+        async_add_entities([UnifiOutletSwitch(obj_id, controller)])
+
+    controller.api.ports.subscribe(async_add_outlet_switch, ItemEvent.ADDED)
+
+    for index in controller.api.outlets:
+        async_add_outlet_switch(ItemEvent.ADDED, index)
```

Comment:
It might make sense to add these with a single `async_add_entites` call since there will be a new task created for each entity with each `async_add_entities` call

https://github.com/home-assistant/core/blob/c6e7b9cc99b01291f1d51c6e8958915b713640d9/homeassistant/helpers/entity_platform.py#L362

Reply:
You don't happen to have a suggested pattern to reference here? :)

Reply:
I don't have a good example.  Maybe make a function that generates the entity from the `object_id` and `controller` and use that in `async_add_outlet_switch` and in the outer scope

Reply:
I will need to fix something generic for multiple platforms, is it Ok if I do a follow-up PR with this?

Reply:
Yes of course, its just a minor inefficiency

Reply:
Resolved here https://github.com/home-assistant/core/pull/80910